src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -323,12 +360,37 @@ public void init(ImportContext context) throws IOException {
     public void init(ImportContext context, boolean prepData) throws IOException {
         context.reattach(catalog);
 
-        ImportData data = context.getData();
-        if (data != null) {
-            addTasks(context, data, prepData); 
+        try {
+            ImportData data = context.getData();
+            if (data != null) {
+                if (data instanceof RemoteData) {
+
+                    data = ((RemoteData) data).resolve(this);
+                    context.setData(data);
+                }
+
+                addTasks(context, data, prepData);
+            }
+
+            // switch from init to pending as needed
+            if (context.getState() == ImportContext.State.INIT) {
+                context.setState(ImportContext.State.PENDING);
+            }
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Failed to resolve remote data", e);
```

Comment:
In another spot I actually put the error message in the Data itself, I guess I can do the same for this one.